### PR TITLE
Added Design pattern/overview

### PR DESCRIPTION
**Pull request by mathanraj0601**

### CODE REVIEW

**Comment by mathanraj0601** [comment]
Make a comment in the line where you can't understand

File: Design Patterns/Overview/Document.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Document structure for every pattern
+
+- Name: name of the pattern
+- Intent: Intent or goal it trying to achieve
+- Alias: Other name for the pattern if exists
+- Motivation/applicability or scenario: The context or scenerio the pattern suits very well.
+- Participant : what are the classes or interface etc., it going to have
+- Collabration : How the participant going to interface with each other
+- Consequences : As every pattern have some tradeoff alomg with it
```
Comment (mksuresh014) [inline Design Patterns/Overview/Document.md]: *along

File: Design Patterns/Overview/Document.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- Implemetation : How to implement in programming language
```
Comment (mksuresh014) [inline Design Patterns/Overview/Document.md]: Implementation

File: Design Patterns/Overview/Document.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+- Name: name of the pattern
+- Intent: Intent or goal it trying to achieve
+- Alias: Other name for the pattern if exists
+- Motivation/applicability or scenario: The context or scenerio the pattern suits very well.
+- Participant : what are the classes or interface etc., it going to have
+- Collabration : How the participant going to interface with each other
+- Consequences : As every pattern have some tradeoff alomg with it
+- Implemetation : How to implement in programming language 
+- Sample code : Sample code for implementation
+- Real time : Is there application using tthe pattern in real world
```
Comment (mksuresh014) [inline Design Patterns/Overview/Document.md]: *the

File: Design Patterns/Overview/Document.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+1. Better and optimal code: Help us write a clean and more optimal code that boosts the performance of the application
+
+## Document structure for every pattern
+
+- Name: name of the pattern
+- Intent: Intent or goal it trying to achieve
+- Alias: Other name for the pattern if exists
+- Motivation/applicability or scenario: The context or scenerio the pattern suits very well.
+- Participant : what are the classes or interface etc., it going to have
+- Collabration : How the participant going to interface with each other
```
Comment (mksuresh014) [inline Design Patterns/Overview/Document.md:25]: not able to understand

File: Design Patterns/Overview/Document.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+- Consequences : As every pattern have some tradeoff alomg with it
+- Implemetation : How to implement in programming language 
+- Sample code : Sample code for implementation
+- Real time : Is there application using tthe pattern in real world
+
+## How to learn pattern
+
+As Experienced people suggest, there are four stages to learn a design pattern or anything
+
+1. Ignorance : Not knowing anything starting from basic
```
Comment (mksuresh014) [inline Design Patterns/Overview/Document.md:35]: *point numbers

File: Design Patterns/Overview/Document.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+- Implemetation : How to implement in programming language 
+- Sample code : Sample code for implementation
+- Real time : Is there application using tthe pattern in real world
+
+## How to learn pattern
+
+As Experienced people suggest, there are four stages to learn a design pattern or anything
+
+1. Ignorance : Not knowing anything starting from basic  
+1. Awakening : knowing about the pattern exist but don't know how to implement (knowing name, indent, Motivation)
```
Comment (mksuresh014) [inline Design Patterns/Overview/Document.md]: *intent

File: Design Patterns/Overview/Document.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+1. OverZealous: Particing the pattern in a controlled environment ( Not in the production code)
+1.  Mastery : After become expert we try to find the pattern fit the application very easily and make better coding
+
+## T shaped learing
+
+there are dozen of design pattern in industry mastery everything is not easy
+So having a awakening stage kind of understanding over wide pattern and having mastery in few pattern helps a alot in software development journey
+
+## Few Pattern to mastery by Experienced people are
+
```
Comment (mksuresh014) [inline Design Patterns/Overview/Document.md:46]: *check numbers

**Review by mksuresh014 — CHANGES_REQUESTED** [review]
Go through the comments

File: Design Patterns/Overview/Document.md (hunk 4 above)
Comment (mathanraj0601) [reply Design Patterns/Overview/Document.md:25]: Please be more specific

File: Design Patterns/Overview/Document.md (hunk 3 above)
Comment (mathanraj0601) [reply Design Patterns/Overview/Document.md]: Readme Ordered list

File: Design Patterns/Overview/Document.md (hunk 4 above)
Comment (mksuresh014) [reply Design Patterns/Overview/Document.md:25]: Collabration : How the participant going to interface with each other  *interface

Comment (mathanraj0601) [reply Design Patterns/Overview/Document.md:25]: >  Collaboration : How the participants are going to interface with each other

Collaboration : How the participants are going to interact with each other

I will correct it and update

**Review by mksuresh014 — CHANGES_REQUESTED** [review]
rectify the mistakes

**Review by mksuresh014 — APPROVED** [review]

**Review by kishore12078 — APPROVED** [review]
Viewed , understood and approved

**Review by ghost — APPROVED** [review]
Viewed and approved

**Review by ajay-kanini — APPROVED** [review]

**Pull request merged**